src/world/mod.rs

```diff
@@ -1,16 +1,106 @@
 pub mod asset;
 pub mod components;
-pub mod json;
+pub mod events;
 
 /// `bevy_ecs_tiled` world related public exports
 pub mod prelude {
     pub use super::asset::*;
     pub use super::components::*;
+    pub use super::events::*;
+    pub use super::WorldChunkedMaps;
 }
 
+use bevy::prelude::*;
+
 use crate::prelude::*;
-use bevy::{asset::RecursiveDependencyLoadState, prelude::*};
-use bevy_ecs_tilemap::prelude::*;
+#[derive(Resource, Default)]
+pub struct WorldChunkedMaps(pub Vec<(Entity, Rect)>);
+
+pub(crate) fn world_chunking(
+    camera: Query<&Transform, (With<Camera>, Changed<Transform>)>,    
+    worlds: Res<Assets<TiledWorld>>,
+    mut commands: Commands,
+    mut world_query: Query<(Entity, &TiledWorldHandle, &TiledWorldSettings, &TiledMapSettings)>,
+    mut chunked_maps: ResMut<WorldChunkedMaps>,
+) {
+    let camera_transform = match camera.iter().next() {
+        Some(camera) => camera,
+        None => return,
+    };
+
+    for (world_entity, world_handle, world_settings, tiled_settings) in world_query.iter_mut() {        
+        if !world_settings.chunking {
+            return;
+        }
+
+        let tiled_world = worlds.get(&world_handle.0).unwrap();
+
+        let offset = match tiled_settings.layer_positioning {
+            LayerPositioning::Centered => Vec3::new(
+                -tiled_world.world_rect.max.x as f32 / 2.0,
+                -tiled_world.world_rect.max.y as f32 / 2.0,
+                0.0,
+            ),
+            LayerPositioning::TiledOffset => Vec3::ZERO,
+        };
+        
+        let tiled_world = worlds.get(&world_handle.0).unwrap();
```

Comment:
Encountered a panic on this line: we could reach this piece of code while the world is not yet fukly loaded.
Please change this to `let Some(tiled_world) = ... else { return; };`

Reply:
Added the option test.